jax/_src/random.py

```diff
@@ -1623,3 +1623,58 @@ def orthogonal(
   q, r = jnp.linalg.qr(z)
   d = jnp.diagonal(r, 0, -2, -1)
   return lax.mul(q, lax.expand_dims(lax.div(d, abs(d).astype(d.dtype)), [-2]))
+
+def gnormal(
+  key: KeyArray,
+  p: float,
+  shape: Sequence[int] = (),
+  dtype: DTypeLikeFloat = dtypes.float_
+) -> jnp.ndarray:
+  """Sample from the generalized normal distribution.
+
+  Args:
+    key: a PRNG key used as the random key.
+    p: a float representing the shape parameter.
+    shape: optional, the batch dimensions of the result. Default ().
+    dtype: optional, a float dtype for the returned values (default float64 if
+      jax_enable_x64 is true, otherwise float32).
+
+  Returns:
+    A random array with the specified shape and dtype.
+  """
+  _check_shape("gnormal", shape)
+  p = core.concrete_or_error(float, p, "The error occurred in jax.random.gnormal()")
```

Comment:
I don't think we need this to be concrete; I believe `gamma` allows dynamic values for its shape parameter.